src/Files.App/DataModels/NavigationControlItems/DriveItem.cs

```diff
@@ -213,10 +213,10 @@ public async Task UpdatePropertiesAsync()
 			}
 		}
 
-		public int CompareTo(INavigationControlItem other)
+		public int CompareTo(INavigationControlItem? other)
 		{
 			var result = Type.CompareTo((other as DriveItem)?.Type ?? Type);
-			return result == 0 ? Text.CompareTo(other.Text) : result;
+			return result == 0 ? (Text is not null ? Text.CompareTo(other?.Text) : 0) : result;
```

Comment:
Writing two lines instead of one here would prevent a double ternary operator, which are a bit of a pain for the clarity of the code.